Запрос на менторство.

nue-01 opened the pull request:
Запрос на менторство.

nue-01 committed:
Запрос на менторство.

nue-01 committed:
Merge branch 'Medusa-master-stream7' into feature/BJS2-40482

flornana commented on src/main/java/school/faang/user_service/dto/mentorshiprequest/MentorshipRequestDto.java:
```diff
@@ -0,0 +1,42 @@
+public class MentorshipRequestDto {
+    private Long id;
+
+    @NotNull()
```
Тут можно указывать сообщение, которое будет пояснять ошибку, например, @NotNull(message = "Requester ID must not be null")

nue-01 replied:
более информативные ошибки я сделал через обработчик улавливающий DataValidationException.
выглядит примерно так, там берется поле и сообщение по умолчанию.
или это не то что ты имела в виду?
![image](https://github.com/user-attachments/assets/9f1e2db4-b2f9-473f-9e5d-71d209f01d04)

flornana replied:
Круто, я также у себя сделала)

flornana commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,33 @@
+        return mentorshipRequestService.getRequests(requestFilterDto);
+    }
+
+    public MentorshipRequestDto acceptRequest(Long requestId) {
```
Стоит поменитить requestId @NotNull либо поменять на маленький long

nue-01 committed:
BJS2-40482-Исправления #1.
Переписал тест связанный с фильтрами в MentorshipRequestServiceImplTest.
Добавил аннотации not null в методах контроллера.

MiraGlaceon commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,34 @@
+public class MentorshipRequestController {
+    private final MentorshipRequestService mentorshipRequestService;
+
+    public MentorshipRequestDto requestMentorship(@Valid MentorshipRequestDto mentorshipRequestDto) {
```
Давай на методы контроллера навешаем аннотаций GetMapping, PostMapping и др. Разберись для чего они нужны и дай краткий комментарий

nue-01 replied:
В http метод запроса это набор зарезервированных слов, которыми клиент дополнительно поясняет что он хочет сделать по отношению к определенному ресурсу.

Принято использовать следующее соответствие между методами и операциями:
POST - Добавление 
PUT - Полное обновление 
PATCH - Частичное обновление 
DELETE - Удаление 
GET - Получение 

При этом реакция на метод прописывается на стороне серверной части, сам http метод ничего не делает а лишь дает понять намерение.

В Spring web есть набор аннотаций, представляющие http методы, которыми мы можем помечать соответствующие методы контроллера.

MiraGlaceon commented on src/main/java/school/faang/user_service/exception/ValidationExceptionHandler.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.Map;
+
+@RestControllerAdvice
+public class ValidationExceptionHandler {
```
За адвайсер большое уважение

nue-01 replied:
Большое спасибо
Правда пока не совсем уверенно им пользуюсь
В основном для того чтобы в postman в случае ошибок было более красивое сообщение

MiraGlaceon replied:
Это отчасти и ответ: обработка ошибок и отдать клиенту в понятной/доступной форме фидбек

nue-01 committed:
BJS2-40482-Исправления #2.
Добавил маппинги в MentorshipRequestController.

nue-01 committed:
Merge branch 'Medusa-master-stream7' into feature/BJS2-40482

nue-01 committed:
BJS2-40482-Исправления #3.
Добавил swagger аннотации.

nue-01 committed:
Merge branch 'Medusa-master-stream7' into feature/BJS2-40482

nue-01 committed:
BJS2-40482-Запрос на менторство.
Убрал дубликаты RejectionDto и UserService; Объединил тесты для UserService; Подправил основную структуру и структуру папок тестов для MentorshipRequest.
